### PR TITLE
Add Template for Win 10 UWP Apps

### PR DESCRIPTION
**Pull request by judgy**
Added a project template for Windows 10 (Universal Windows Plattform)
Removed not needed file 'PrismUnityApp.csproj' (See
https://msdn.microsoft.com/en-us/library/ms185308.aspx)

### CODE REVIEW

**Comment by brianlagunas** [comment]
The PrismUntyApp.csproj is required.  It's the VS project for creating the template in the TemplatePack solution.  Also, UWP can't use the Prism.Unity NuGet package, it has to use the new Prism.Unity.Forms packages.

**Comment by judgy** [comment]
1.) The ProjectTemplate is still working without the PrismUntyApp.csproj.
I haven't found any ProjectTemplate with a root csproj file anywhere - and no documentation for that.
2.) I've just found the new NuGet package Prism.Unity.Forms. (The prerelease has been downloaded just 32 times..). I'll replace it.
3.) The WinPhone8 project is still legacy, isn't it?
Even if it is still part of the original Xamarin template, it should be removed.

**Comment by brianlagunas** [comment]
No, it's not working fine without it because the PrismUntyApp.csproj file is what the PrismTemplatePack.sln needs to add the project to the solution.

For now, we will not be removing any projects from the XF app template.  If devs don't need it, they can remove it.  When Xamarin has officially removed it from their templates, then we will follow suite.

**Comment by judgy** [comment]
Sorry, haven't seen the file *PrismTemplatePack.sln* in the folder *\Extensibility\TemplatePack\*.
Have also exchanged Prism.Unity.Forms.

This is the first time I'm sending PRs. So I hope, I've done it in the correct order.
Please go on doing this great job. 
I'm supporting you by watching all your Pluralsight courses..

File: Extensibility/TemplatePack/ProjectTemplates/CSharp/Forms/PrismUnityApp/UnityApp.UWP/UnityAppTemplate.UWP.csproj [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      <Generator>MSBuild:Compile</Generator>
+      <SubType>Designer</SubType>
+    </Page>
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\$saferootprojectname$\$saferootprojectname$.csproj">
+      <Name>$saferootprojectname$</Name>
+    </ProjectReference>
+  </ItemGroup>
+  <ItemGroup>
```
Comment (brianlagunas) [inline Extensibility/TemplatePack/ProjectTemplates/CSharp/Forms/PrismUnityApp/UnityApp.UWP/UnityAppTemplate.UWP.csproj]: delete this item group

File: Extensibility/TemplatePack/ProjectTemplates/CSharp/Forms/PrismUnityApp/UnityApp.UWP/UnityApp.UWP.vstemplate [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    <FullClassName>Microsoft.VisualStudio.Universal.TemplateWizards.PlatformVersion.Wizard</FullClassName>
+  </WizardExtension>
+  <WizardData>
+    <packages repository="registry" keyName="NETCoreSDK" isPreunzipped="true">
+      <package id="Microsoft.NETCore.UniversalWindowsPlatform" version="5.0.0" skipAssemblyReferences="false" />
+    </packages>
+    <packages repository="extension" repositoryId="Prism.TemplatePack.138e3411-ac97-4c11-8016-c27a831cba2e">
+      <package id="Xamarin.Forms" version="2.0.0.6490" />
+      <package id="Prism.Core" version="6.1.0" />
+      <package id="Prism.Forms" version="6.1.0-pre2" />
```
Comment (brianlagunas) [inline Extensibility/TemplatePack/ProjectTemplates/CSharp/Forms/PrismUnityApp/UnityApp.UWP/UnityApp.UWP.vstemplate]: wrong version of Prism.Forms.  Prism.Unity.Forms depends on Prism.Form 6.1.0-pre-3.

File: Extensibility/TemplatePack/ProjectTemplates/CSharp/Forms/PrismUnityApp/UnityApp.UWP/project.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+﻿{
+  "dependencies": {
+    "Microsoft.ApplicationInsights": "1.0.0",
```
Comment (brianlagunas) [inline Extensibility/TemplatePack/ProjectTemplates/CSharp/Forms/PrismUnityApp/UnityApp.UWP/project.json]: only the default dependencies should be placed in here:

```
{
  "dependencies": {
    "Microsoft.NETCore.UniversalWindowsPlatform": "5.0.0"
  },
  "frameworks": {
    "uap10.0": {}
  },
  "runtimes": {
    "win10-arm": {},
    "win10-arm-aot": {},
    "win10-x86": {},
    "win10-x86-aot": {},
    "win10-x64": {},
    "win10-x64-aot": {}
  }
}
```

File: Extensibility/TemplatePack/ProjectTemplates/CSharp/Forms/PrismUnityApp/UnityApp.UWP/UnityApp.UWP.vstemplate [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            <ProjectItem ReplaceParameters="false" TargetFileName="Square150x150Logo.scale-200.png">Square150x150Logo.scale-200.png</ProjectItem>
+            <ProjectItem ReplaceParameters="false" TargetFileName="Square44x44Logo.scale-200.png">Square44x44Logo.scale-200.png</ProjectItem>
+            <ProjectItem ReplaceParameters="false" TargetFileName="Square44x44Logo.targetsize-24_altform-unplated.png">Square44x44Logo.targetsize-24_altform-unplated.png</ProjectItem>
+            <ProjectItem ReplaceParameters="false" TargetFileName="StoreLogo.png">StoreLogo.png</ProjectItem>
+            <ProjectItem ReplaceParameters="false" TargetFileName="Wide310x150Logo.scale-200.png">Wide310x150Logo.scale-200.png</ProjectItem>
+      </Folder>
+      <ProjectItem ReplaceParameters="false" TargetFileName="$projectname$_TemporaryKey.pfx" BlendDoNotCreate="true">Application_TemporaryKey.pfx</ProjectItem>
+    </Project>
+  </TemplateContent>
+  <WizardExtension>
```
Comment (brianlagunas) [inline Extensibility/TemplatePack/ProjectTemplates/CSharp/Forms/PrismUnityApp/UnityApp.UWP/UnityApp.UWP.vstemplate]: What are the purpose of all these added template wizards?  Do you know there function and why they are needed?

**Comment by brianlagunas** [comment]
I added some notes for your review.

**Comment by brianlagunas** [comment]
There is going to be a large change to the Xamarin.Forms project template, so I am going to close this.  This PR is great, but I have so much work to do, that it would be easier for me just to include it myself as I modify the other parts of the template.  This also requires downloading the dependent Nuget apckages locally to be deployed with the VSIX.

**Comment by judgy** [comment]
Hi Brian, ok, just updated the template. Hadn't the time to test and commit the changes yet. So I'll commit it only in my prism fork. 
If you want to do it yourself, it's fine. Thought you could need some help.
I'm also developing some T4 templates to create complete Prism/Xamarin solutions.

**Comment by brianlagunas** [comment]
I am going to use your template, but since I have to change the template altogether, I figured I would avoid merging since I have to make so many changes.

**Comment by judgy** [comment]
Ok, I've commited all updates to
https://github.com/judgy/Prism/tree/master/Extensibility/TemplatePack
Feel free to take what you need.
To your comments above:
You cannot remove all items you commented above.
For example you need the reference to the portable class.
You can only remove the ApplicationInsight Wizzards.
Something I still don't know: How can the template be modified so that the Configuration Manager of VS has the 'build' and 'deploy' checkbox marked.
Have fun on the Evolve16

**Comment by brianlagunas** [comment]
Great! Thanks for the help.  I've already download your repo locally.

**Pull request closed without merging**